### PR TITLE
Show partially complete error

### PR DESCRIPTION
**Pull request by MancunianSam**
- Make consignment reference not optional
- Bump to latest version of generated graphql
- Add upload in progress page and messages
- Add new endpoint and route
- Add consigment status service and tests
- Add in consignment status check to action
- Add new tests for the upload endpoint

### CODE REVIEW

File: test/services/ConsignmentStatusServiceSpec.scala
Comment (TomJKing) [inline test/services/ConsignmentStatusServiceSpec.scala:33]: Maybe add the word "function" to make it clear what is being tested: `"consignmentStatus function" should ...`

**Review by TomJKing — COMMENTED** [review]
Looks good. A few minor comments.

I think these changes will break the e2e tests, so these will need updating as well.

Also when trying to run locally against intg I was getting the following error. Does the DB need updating?

`Unexpected response from GraphQL API: Response(Left({"data":null,"errors":[{"message":"Cannot query field 'currentStatus' on type 'Consignment'. (line 1, column 95):\nquery getConsignmentStatus($consignmentId:UUID!){getConsignment(consignmentid:$consignmentId){currentStatus{upload}}}\n                                                                                              ^","locations":[{"column":95,"line":1}]}]}),400,Bad Request ... `

File: app/validation/ValidatedActions.scala
Comment (MancunianSam) [reply app/validation/ValidatedActions.scala:28]: I didn't yet because the tests in `UploadControllerSpec` cover the different paths through this method so I didn't think we needed tests for this method specifically. I'm happy to put some in but that's why I didn't originally. What do you think?

File: conf/messages
Comment (MancunianSam) [reply conf/messages:64]: Yep, I think so, I've added one in.

**Comment by MancunianSam** [comment]
The API is failing because there's a PR still open for the changes to add that field. Once that's merged, this should work but I can't merge this until then.

File: app/validation/ValidatedActions.scala
Comment (TomJKing) [reply app/validation/ValidatedActions.scala:27]: I'd rather see separate unit tests, as don't really like relying on tests for another class to check logic from another class, can lead to issues when things get more complicated in the future.

Comment (MancunianSam) [reply app/validation/ValidatedActions.scala:27]: Yeah good point. I've added in some tests for this method and I added in a couple for the existing consignmentExists method as well. I think we're going to start using this as some kind of central validation point so the tests will be useful.

**Comment by MancunianSam** [comment]
The API changes are in now so this can me merged when it's ready.

**Comment by MancunianSam** [comment]
You're right about the e2e tests as well, there was one that was trying to upload twice which you can't do with this change. There's a PR here. https://github.com/nationalarchives/tdr-e2e-tests/pull/101/files

File: app/validation/ValidatedActions.scala
Comment (suzannehamilton) [inline app/validation/ValidatedActions.scala:41]: This is interesting - I expected us to show the same page with a different message rather than doing it as a redirect.

What's the reason for doing it this way? I haven't thought about the trade-offs in any detail, so maybe this is the best option. My initial thought is that it's a bit odd to have a separate page because do you then have to redirect _back_ if someone happens to go there after the upload has finished?

Comment (MancunianSam) [reply app/validation/ValidatedActions.scala:41]: The reason I did it as a separate page is because the upload page is getting quite complicated already with different bits being shown or hidden depending on the state. This make accessibility more difficult as you have to make sure the page isn't dependant on css to show or hide things and we've already got bugs with things not showing when they should so I didn't want to make it worse. 
You're right about redirecting back when the upload is complete but we can add this logic into the controllers and it should be a bit clearer what's happening rather than digging through the javascript to see which bits are shown/hidden.

Comment (suzannehamilton) [reply app/validation/ValidatedActions.scala:41]: Wait, wouldn't this logic be in the controller or HTML templates rather than the JavaScript and CSS?

This controller won't be any more complex if you render a different template on line 41 rather than do a redirect, and the user journeys will be easier to think about too because we don't need to add redirects on both controller actions.

Comment (MancunianSam) [reply app/validation/ValidatedActions.scala:41]: You're right about the html templates, ignore what I said about the css. I think I see what you mean. I thought you were talking about adding more if/else logic to the existing `upload.scala.html` which is what I was trying to avoid but do you mean rendering the `uploadInProgress.scala.html` template inside the validated actions? Like this 
```scala
if(!transferAgreementExists) {
        Redirect(routes.TransferAgreementController.transferAgreement(consignmentId))
      } else if(uploadInProgress) {
        Ok(views.html.uploadInProgress(consignmentId))
      } else {
        f(request)
      }
``` 
I've changed it to this as it makes sense I think

Comment (suzannehamilton) [reply app/validation/ValidatedActions.scala]: Great, thanks!

**Comment by suzannehamilton** [comment]
I haven't fully reviewed this - I just commented on the controller because I noticed the redirect in the end-to-end test. @TomJKing are you happy with this PR? I'll let you approve it if so.

**Review by TomJKing — APPROVED** [review]

**Pull request merged**
